code/game/objects/effects/anomalies.dm

```diff
@@ -419,6 +421,82 @@
 			if(EXPLODE_LIGHT)
 				SSexplosions.lowturf += T
 
+/obj/effect/anomaly/delimber
+	name = "delimber anomaly"
+	icon_state = "delimber_anomaly"
+	aSignal = /obj/item/assembly/signaler/anomaly/delimber
+	immortal = TRUE
+	/// Time passed since the last effect, increased by delta_time of the SSobj
+	var/ticks = 0
+	/// How many seconds between each anomaly pulses
+	var/releasedelay = 15
+	/// Range of the anomaly pulse
+	var/range = 5
+	///Lists for zones and bodyparts to swap and randomize
+	var/static/list/zones = list(BODY_ZONE_HEAD, BODY_ZONE_CHEST, BODY_ZONE_L_ARM, BODY_ZONE_R_ARM, BODY_ZONE_L_LEG, BODY_ZONE_R_LEG)
+	var/static/list/chests
+	var/static/list/heads
+	var/static/list/l_arms
+	var/static/list/r_arms
+	var/static/list/l_legs
+	var/static/list/r_legs
+	var/static/list/organs
+
+/obj/effect/anomaly/delimber/Initialize(mapload, new_lifespan, drops_core)
+	. = ..()
+	if(!chests)
+		chests = subtypesof(/obj/item/bodypart/chest)
+	if(!heads)
+		heads = subtypesof(/obj/item/bodypart/head)
+	if(!l_arms)
+		l_arms = subtypesof(/obj/item/bodypart/l_arm)
+	if(!r_arms)
+		r_arms = subtypesof(/obj/item/bodypart/r_arm)
+	if(!l_legs)
+		l_legs = subtypesof(/obj/item/bodypart/l_leg)
+	if(!r_legs)
+		r_legs = subtypesof(/obj/item/bodypart/r_leg)
+	if(!organs)
+		organs = subtypesof(/obj/item/organ)
+
+/obj/effect/anomaly/delimber/anomalyEffect(delta_time)
+	. = ..()
+	ticks += delta_time
+	if(ticks < releasedelay)
+		return
+	ticks -= releasedelay
+
+	swap_parts(range)
+
+/obj/effect/anomaly/delimber/proc/swap_parts(swap_range)
+	for(var/mob/living/carbon/nearby in range(swap_range, src))
+		if(nearby.run_armor_check(attack_flag = BIO, absorb_text = "Your armor protects you from [src]!") >= 100)
+			continue //We are protected
+		var/picked_zone = pick(zones)
+		var/obj/item/bodypart/picked_user_part = nearby.get_bodypart(picked_zone)
```

Comment:
get_bodypart() can return null if the target is missing said bodypart.

Reply:
i'll add a check later